wgengine/magicsock/magicsock.go

```diff
@@ -5105,6 +5125,88 @@ func (s derpAddrFamSelector) PreferIPv6() bool {
 	return false
 }
 
+type endpointTrackerEntry struct {
+	endpoint tailcfg.Endpoint
+	until    time.Time
+}
+
+type endpointTracker struct {
+	mu    sync.Mutex
+	cache map[netip.AddrPort]endpointTrackerEntry
+}
+
+const endpointTrackerLifetime = 5*time.Minute + 10*time.Second // slightly longer than a full netcheck period
+
+func (et *endpointTracker) update(now time.Time, eps []tailcfg.Endpoint) (ret []tailcfg.Endpoint) {
```

Comment:
document what it returns? Or change `ret` to something like `stillAlive` or `newlyAdded` or `nowDeleted` or whatever it is

Reply:
Renamed to `epsPlusCached`